### ISSUE TITLE
Theoretical - How UMAP consider labels in the train and test, and how UMAP learn in train 

### ISSUE THREAD

**Issue by alon1samuel** [lmcinnes/umap#540]
Attention - theoretical question. Feel free to close if its not beneficial with the repo. 

I was reading on UMAP and used it for decreasing the dimension of my dataset. I understood the iterative process that minimizes the Cross-Entropy between the high and low dimensional spaces.

1 - I couldn't find someone talking about how UMAP using labels when applying the algorithm of building the probability matrix or the iterative optimization part?

2 - I didn't understand how UMAP learn something from the train set and then infer it to the test set. From my understanding, it takes the train set from the high dimensional space and project it on the low dimensional space. If a test set comes along then it has to learn the distances all over again, no?

**Comment by alon1samuel** [comment]
@lmcinnes Thanks for the answer, you answered it for me in private. 
I'm adding your answer if that's ok. For anyone else that have the same thoughts:

Hi Alon,

For supervised learning UMAP uses the labels as a distinct dataset with a novel metric (a categorical distance). It generates a fuzzy simplicial set for the label data under that metric and then intersects that with the fuzzy simplicial set for the base data (with some corrections for UMAP assumptions such as local connectivity). This gives a combined fuzzy simplicial set (that can be embedded in the same way as UMAP usually does) that combines information from the base data as well as label information.

For inference the trick is to add the new samples to the existing fuzzy simplicial set, and then re-optimize the embedding of the new larger fuzzy simplicial set under the constraint that only the newly added samples are allowed to move during the optimization.

If you are making use of these techniques you may also want to look into the new support for ParametricUMAP that has been added recently (available on github, but not in the standard release yet). https://umap-learn.readthedocs.io/en/latest/parametric_umap.html

Leland.
